meteor/client/ui/RundownList/RundownListItem.tsx

```diff
@@ -0,0 +1,278 @@
+		 * In addition it also should not be handled if the rundown is part of a
+		 * playlist with multiple rundowns, because then the playlist component
+		 * will be the legal drop target and handle the drop itself.
+		 */
+		return (
+			Settings.allowMultiplePlaylistsInGUI === true &&
+			props.isOnlyRundownInPlaylist === true &&
+			props.playlistId !== undefined &&
+			monitor.getItemType() === RundownListDragDropTypes.RUNDOWN
+		)
+	},
+
+	drop(props: IRundownListItemProps, monitor: DropTargetMonitor) {
+		if (monitor.didDrop()) {
+			return
+		}
+
+		const dropped = monitor.getItem()
+
+		console.debug(`Drop on rundown ${props.rundown._id} (playlist ${props.playlistId}):`, dropped)
+
+		if (isRundownDragObject(dropped)) {
+			return {
+				type: 'HANDLE_RUNDOWN_DROP',
+				rundownId: dropped.id,
+				targetPlaylistId: props.playlistId,
+			}
+		}
+	},
+
+	hover(props: IRundownListItemProps, monitor: DropTargetMonitor) {
+		if (monitor.getItemType() === RundownListDragDropTypes.RUNDOWN) {
+			const item = monitor.getItem() as IRundownDragObject
+
+			// if that rundown is not this rundown
+			if (item && props.rundown._id !== item.id) {
+				props.swapRundownOrder(props.rundown._id, item.id)
+			}
+		}
+	},
+}
+
+const dropCollect: DropTargetCollector<IRundownDropTargetProps, IRundownListItemProps> = (
+	connect: DropTargetConnector,
+	monitor: DropTargetMonitor,
+	props: IRundownListItemProps
+) => {
+	return {
+		connectDropTarget: connect.dropTarget(),
+	}
+}
+
+interface IRundownDragLayerProps {
+	currentOffset: XYCoord | null
+	clientOffset: XYCoord | null
+}
+
+const dragLayerCollect: DragLayerCollector<
+	IRundownDragSourceProps & IRundownListItemProps,
+	IRundownDragLayerProps
+> = function(monitor, props) {
+	let currentOffset: XYCoord | null = null
+	let clientOffset: XYCoord | null = null
+
+	if (monitor.getItem()?.id === props.rundown._id) {
+		currentOffset = monitor.getDifferenceFromInitialOffset()
+		clientOffset = monitor.getClientOffset()
+	}
+
+	return {
+		currentOffset,
+		clientOffset,
+	}
+}
+
+export const RundownListItem = withTranslation()(
+	// DragLayer(dragLayerCollect, {
+	// 	arePropsEqual: (props, newProps) => {
+	// 		if (props.rundown._id !== newProps.rundown._id) {
+	// 			return false
+	// 		}
+
+	// 		return true
+	// 	},
+	// })(
+	DragSource(
+		RundownListDragDropTypes.RUNDOWN,
+		dragSpec,
+		dragCollector
+	)(
+		DropTarget(
+			RundownListDragDropTypes.RUNDOWN,
+			dropSpec,
+			dropCollect
+		)(
+			class RundownListItem extends React.Component<
+				Translated<IRundownListItemProps> & IRundownDragSourceProps & IRundownDropTargetProps
+			> {
+				studio: Studio
+				showStyle: ShowStyleBase
+
+				constructor(props: Translated<IRundownListItemProps> & IRundownDragSourceProps & IRundownDropTargetProps) {
+					super(props)
+
+					this.studio = this.props.rundown.getStudio()
+					this.showStyle = this.props.rundown.getShowStyleBase()
```

Comment:
Sometimes, when starting up, the list view crashes with this message, related to this line:
![obraz](https://user-images.githubusercontent.com/3635991/98752865-5fee9c00-23c3-11eb-957d-9cd44db2cdfd.png)

This needs to be changed, so that the component can re-render once the ShowStyleBase streams into the browser/isn't in the local collection yet.

Reply:
Maybe the RundownListItem could use a tracker? or we could inject these two properties in the parent-level tracker?

Reply:
Now that we've separated the item and the view of it into two components already I guess that adding a tracker to the data component won't make it too complicated?